src/tdyutils.c

```diff
@@ -482,6 +482,31 @@ PetscInt TDyReturnIndexInList(PetscInt *list, PetscInt nlist, PetscInt value) {
 }
 
 /* -------------------------------------------------------------------------- */
+/// Outputs a sequential PETSc Vec as binary
+///
+/// @param [in] vec A sequential PETSc Vec
+/// @param [in] filename Name of the file
+/// @returns 0 on success, or a non-zero error code on failure
+PetscErrorCode TDySavePetscVecSeqAsBinary(Vec vec, const char filename[]) {
```

Comment:
I don't think we need this function. Instead, we could modify `TDySavePetscVecAsBinary` to get the communicator for the given Vec and use it instead of `PETSC_COMM_WORLD` there. Then it would work regardless of the type of Vec.

Let me know if you'd like me to do this.

Reply:
I agree it should be deleted.